### PR TITLE
trie/verkle: change tree key generation

### PR DESCRIPTION
**Pull request by jsign**
This PR changes how we transform a point to bytes to be used as a tree key.

~Note: still in draft since it's waiting for a small PR merge in go-verkle, but it's ready for review.~
Note: we should merge this PR as soon as the target branch is ready for the new testnet.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -14,19 +14,19 @@ require (
 	github.com/cloudflare/cloudflare-go v0.14.0
 	github.com/cockroachdb/pebble v0.0.0-20230209160836-829675f94811
 	github.com/consensys/gnark-crypto v0.12.1
-	github.com/crate-crypto/go-ipa v0.0.0-20231025140028-3c0104f4b233
+	github.com/crate-crypto/go-ipa v0.0.0-20240223125850-b1e8a79f509c
```
Comment (jsign) [inline go.mod:17]: For now, I'm sitting on top of https://github.com/ethereum/go-verkle/pull/429, but when that is merged I'll change it to master.

File: trie/utils/verkle.go [hunk 2]
```diff
@@ -187,15 +187,7 @@ func GetTreeKeyCodeChunkWithEvaluatedAddress(addressPoint *verkle.Point, chunk *
...
-	// The output of Byte() is big engian for banderwagon. This
-	// introduces an imbalance in the tree, because hashes are
-	// elements of a 253-bit field. This means more than half the
-	// tree would be empty. To avoid this problem, use a little
-	// endian commitment and chop the MSB.
-	retb := evaluated.Bytes()
-	for i := 0; i < 16; i++ {
-		retb[31-i], retb[i] = retb[i], retb[31-i]
-	}
+	retb := verkle.HashPointToBytes(evaluated)
```
Comment (jsign) [inline trie/utils/verkle.go:190]: For more info: https://github.com/ethereum/go-verkle/pull/428

**Review by kevaundray — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (jsign) [reply go.mod:17]: Done

**Review by gballet — APPROVED** [review]
LGTM

**Pull request merged**